In [3]:
%pip install hummingbird-ml

^C
Note: you may need to restart the kernel to use updated packages.


  Using cached hummingbird_ml-0.4.12-py2.py3-none-any.whl.metadata (11 kB)
  Using cached onnxconverter_common-1.14.0-py2.py3-none-any.whl.metadata (4.2 kB)
  Using cached onnx-1.16.1-cp312-cp312-win_amd64.whl.metadata (16 kB)
  Using cached scipy-1.15.2-cp312-cp312-win_amd64.whl.metadata (60 kB)
  Using cached scikit_learn-1.6.1-cp312-cp312-win_amd64.whl.metadata (15 kB)
  Using cached torch-2.7.0-cp312-cp312-win_amd64.whl.metadata (29 kB)
  Using cached dill-0.4.0-py3-none-any.whl.metadata (10 kB)
  Using cached filelock-3.18.0-py3-none-any.whl.metadata (2.9 kB)
  Using cached networkx-3.4.2-py3-none-any.whl.metadata (6.3 kB)
  Using cached jinja2-3.1.6-py3-none-any.whl.metadata (2.9 kB)
  Using cached fsspec-2025.3.2-py3-none-any.whl.metadata (11 kB)
  Using cached MarkupSafe-3.0.2-cp312-cp312-win_amd64.whl.metadata (4.1 kB)
  Using cached joblib-1.5.0-py3-none-any.whl.metadata (5.6 kB)
Using cached hummingbird_ml-0.4.12-py2.py3-none-any.whl (166 kB)
Using cached onnx-1.16.1-cp312-c

In [5]:
%pip install numpy
%pip install scikit-learn

Note: you may need to restart the kernel to use updated packages.
Note: you may need to restart the kernel to use updated packages.


In [4]:
import numpy as np
from sklearn.ensemble import RandomForestClassifier
from hummingbird.ml import convert, load

In [23]:
records = 5*100000

In [24]:
# Create some random data for binary classification
num_classes = 2
X = np.random.rand(records, 28)
y = np.random.randint(num_classes, size=records)

In [25]:
# Create and train a model (scikit-learn RandomForestClassifier in this case)
skl_model = RandomForestClassifier(n_estimators=20, max_depth=10, n_jobs=-1)

In [26]:
skl_model.fit(X, y)


RandomForestClassifier(max_depth=10, n_estimators=20, n_jobs=-1)

In [27]:
# Use Hummingbird to convert the model to PyTorch
model = convert(skl_model, 'pytorch')

In [30]:
# Run predictions on CPU
model.predict(X)

array([1, 1, 0, ..., 1, 1, 0], shape=(500000,))

In [31]:
# Run predictions on GPU
model.to('cuda')
model.predict(X)

AssertionError: Torch not compiled with CUDA enabled

In [ ]:
# Save the model
model.save('hb_model')

In [ ]:
# Load the model back
model = load('hb_model')

conda create -n rapids-23.06 -c rapidsai -c nvidia -c conda-forge cuml=23.06 python=3.10 cudatoolkit=11.4


pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118


conda install pytorch torchvision torchaudio pytorch-cuda=11.4 -c pytorch -c nvidia
